charts/external-dns/templates/_helpers.tpl

```diff
@@ -70,3 +70,30 @@ The image to use
 {{- define "external-dns.image" -}}
 {{- printf "%s:%s" .Values.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.image.tag) }}
 {{- end }}
+
+
+{{/*
+Keep backward compatibility on provider
+*/}}
+{{- define "external-dns.providername" -}}
+{{- if eq (typeOf .Values.provider) "string" }}
+{{- .Values.provider }}
+{{- else }}
+{{- .Values.provider.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+The webhook providers
+*/}}
+{{- define "external-dns.webhook-providers" -}}
+{{- if eq (typeOf .Values.provider) "object" }}
+{{- $version := default "latest" .Values.provider.version }}
+adguard: ghcr.io/muhlba91/external-dns-provider-adguard:{{ $version }}
+ionos: ghcr.io/ionos-cloud/external-dns-ionos-webhook:{{ $version }}
+stackit: ghcr.io/stackitcloud/external-dns-stackit-webhook:{{ $version }}
+{{- if eq .Values.provider.name "custom" }}
```

Comment:
This is unnecessarily complicated. It should just take the image in `provider.name`, handling it as an image if it has a `/` character in it.

Reply:
I disagree. Words has meanings. A name is not an docker image url. An image is not the name of a project.
The name of this project is not `registry.k8s.io/external-dns/external-dns`.
The image of this project is not `external-dns`.

Reply:
In essence, one is specifying the provider to use. The provider can be either an image or a built-in.

Separate config settings for the built-in and image cases would be reasonable, but people seemed to object to that in my PR.

Having to modify the chart to add to a built-in-chart lookup table is just too much. And one shouldn't have to change two settings to use an image provider.

Reply:
I removed the lookup table.
The provider has a name in external dns, which is not a docker image url.